README.md

```diff
@@ -1,2 +1,54 @@
-# rtx-kg2-gateway
+# RTX-KG2 Gateway
+
 Enabling RTX-KG2 data access through various means.
+
+## Installation
+
+Please use Python [`poetry`](https://python-poetry.org/) to run and install related content.
+The Poetry environment for this project includes dependencies which help run IDE environments, manage the data, and run workflows.
+
+```bash
+# after installing poetry, create the environment
+poetry install
+```
+
+## Development
+
+### Jupyter Lab
+
+Please follow installation steps above and then use a relevant Jupyter environment to open and explore the notebooks under the `notebooks` directory.
+
+```bash
+# after creating poetry environment, run jupyter
+poetry run jupyter lab
+```
```

Comment:
I'm interpreting this as a "how to get started using the data" section, so you might consider changing the title from "Jupyter Lab" to "Querying the Data" or something along those lines.

Good that you show how to start Jupyter Lab! You might consider adding a short tutorial where you query the data, e.g. drawing their attention to a particular notebook where they can start trying queries and any setup they might need to do before running their first query. In the tutorial, you might even show a sample query, its result, and how you can do things with the result (e.g., drawing a graph of the resulting nodes).

Not necessary IMO, but you might consider giving a simple high-level overview of what Kuzu is doing, e.g. that it's creating an in-memory database on which to perform Cypher queries on the RTX-KG2 graph.

Reply:
Also, I see you're using Jupytext to sync the `.py` and `.ipynb` files; is that something that gets done automatically as they edit the `.ipynb` files in Jupyter Lab, or do they have to manually invoke them? Would that also occur if they used, say, VS Code's notebook interface? I ask because I know some people in the Greene Lab use VS Code with notebooks.

Generally, it'd be good to give them some direction on how to maintain the link between the notebook and source files if there's anything they have to do.

Reply:
Thanks, great thoughts! Jupyter Lab was explicitly used to create the `.py` pair files through [`jupytext`](https://jupytext.readthedocs.io/en/latest/) and related configuration within the `pyproject.toml` file. Using Jupyter Lab through the Poetry environment this way enables one to have roughly the same reproducible IDE environment I used to create and run things (outside of Poe the Poet).

Development was performed through Jupyter Lab as a fully open source platform to help integrate with [extensions](https://jupyterlab.readthedocs.io/en/stable/user/extensions.html) like `jupytext` (this also especially applies to advanced visualization technologies like `plotly` which enable notebook renderings which may otherwise experience issues).

Acknowledging I could be very wrong, and being open to learning more, my personal experience with VS Code and Jupyter notebook files is that it sometimes results in reproducibility challenges due to undefined (i.e. no or very little source code tracking for) differences with VS Code version, configurations, and (VS Code-specific) extensions. Jupyter Lab is coupled to the reproducible execution of notebooks. Perhaps it could be understood as the default authoritative "notebook interpreter" just as we sometimes assume CPython as the default authoritative Python interpreter. That said, I feel it's best to be included as part of the Python environment instead of opting for external and less certain processing (through VS Code).

I'll work towards adding some content to this effect, as you're right about all of it - it needs more guidance. Generally I think there's more to consider here too concerning Google Colab and WASM-focused platforms like [`jupyterlite`](https://github.com/jupyterlite/jupyterlite).

Reply:
I've added a bit more clarity here with gentle guidance under "Running and updating Jupyter notebooks". For example work I've opened a new issue where I hope to focus on this a bit more specifically.